### PR TITLE
feat!: `test` statements

### PR DESCRIPTION
**Pull request by Chriscbr**
Wing now has a simpler way of writing unit tests: the `test` statement:

```js
let b = new cloud.Bucket();

test "put" {
  b.put("hello.txt", "world");
  assert(b.list().length == 1);
}
```

A test accepts a block of inflight code, and can be executed with `wing test`. Tests will be executed in isolated arguments, so operations that affect stateful resources (like buckets, queues, etc.) will not affect the results of each other.

BREAKING CHANGES:
* Removed `listTests`, `runTest`, and `runAllTests` from the `Simulator` class in `@winglang/sdk`. 

TODO:
- [x] update spec
- [x] update examples
- [x] syntax highlight "test" keyword

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Monada Contribution License](https://docs.winglang.io/terms-and-policies/contribution-license.html)*.

### CODE REVIEW

File: apps/wing/src/commands/test.ts [hunk 1]
```diff
@@ -161,9 +161,11 @@ async function testTfAws(synthDir: string): Promise<sdk.cloud.TestResult[]> {
 
   const [testRunner, tests] = await withSpinner("Setting up test runner...", async () => {
     const testArns = await terraformOutput(synthDir, ENV_WING_TEST_RUNNER_FUNCTION_ARNS);
+    console.error(JSON.stringify(testArns, null, 2));
```
Comment (Chriscbr) [inline apps/wing/src/commands/test.ts:164]: Clean up

File: libs/wingc/src/ast.rs [hunk 2]
```diff
@@ -447,6 +447,10 @@ pub enum StmtKind {
 		catch_block: Option<CatchBlock>,
 		finally_statements: Option<Scope>,
 	},
+	// Test {
+	// 	name: Symbol,
+	// 	statements: Scope,
+	// },
```
Comment (Chriscbr) [inline libs/wingc/src/ast.rs:453]: Remove

File: libs/wingsdk/src/cloud/test.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+/**
+ * Inflight client for `ITestHandler`.
+ */
+export interface ITestHandlerClient {
+  /**
+   * Entrypoint function that will be called when the test is run.
+   * @inflight
+   */
+  handle(event: string): Promise<void>;
```
Comment (Chriscbr) [inline libs/wingsdk/src/cloud/test.ts:83]:
```suggestion
  /**
   * Inflight function that will be called when the test is run.
   * @inflight
   */
  handle(): Promise<void>;
```

File: tools/hangar/__snapshots__/test_corpus/api.w_test_sim.md [hunk 4]
```diff
@@ -4,6 +4,6 @@
 ```log
 - Compiling to sim...
 ✔ Compiling to sim...
-pass ─ api.wsim » root/env0/test
```
Comment (eladb) [inline tools/hangar/__snapshots__/test_corpus/api.w_test_sim.md:7]: Regression?

File: tools/hangar/__snapshots__/test_corpus/api_path_vars.w_test_sim.md [hunk 5]
```diff
@@ -4,6 +4,6 @@
 ```log
 - Compiling to sim...
 ✔ Compiling to sim...
-pass ─ api_path_vars.wsim » root/env0/test
+pass ─ api_path_vars.wsim » root/env0/test:test
```
Comment (eladb) [inline tools/hangar/__snapshots__/test_corpus/api_path_vars.w_test_sim.md:7]: can we special-case this to be "test"?

File: tools/hangar/__snapshots__/test_corpus/capture_containers.w_test_sim.md [hunk 6]
```diff
@@ -4,6 +4,6 @@
 ```log
 - Compiling to sim...
 ✔ Compiling to sim...
-pass ─ capture_containers.wsim » root/env0/test
+pass ─ capture_containers.wsim (no tests)
```
Comment (eladb) [inline tools/hangar/__snapshots__/test_corpus/capture_containers.w_test_sim.md:7]: regression?

File: tools/hangar/__snapshots__/test_corpus/capture_containers.w_compile_tf-aws.md [hunk 7]
```diff
@@ -21,7 +21,7 @@
   },
   "output": {
     "WING_TEST_RUNNER_FUNCTION_ARNS": {
-      "value": "[[\"root/Default/Default/test\",\"${aws_lambda_function.root_test_AAE85061.arn}\"]]"
+      "value": "[]"
```
Comment (eladb) [inline tools/hangar/__snapshots__/test_corpus/capture_containers.w_compile_tf-aws.md:24]: regression?

File: tools/hangar/__snapshots__/test_corpus/capture_in_binary.w_test_sim.md [hunk 8]
```diff
@@ -4,6 +4,6 @@
 ```log
 - Compiling to sim...
 ✔ Compiling to sim...
-pass ─ capture_in_binary.wsim » root/env0/test
+pass ─ capture_in_binary.wsim (no tests)
```
Comment (eladb) [inline tools/hangar/__snapshots__/test_corpus/capture_in_binary.w_test_sim.md:7]: regression?

File: docs/06-contributors/999-rfcs/000-stories/story-12b.md
Comment (eladb) [inline docs/06-contributors/999-rfcs/000-stories/story-12b.md:1]: revert - don't change the past

**Review by eladb — APPROVED** [review]

File: tools/hangar/__snapshots__/test_corpus/api_path_vars.w_test_sim.md (hunk 5 above)
Comment (Chriscbr) [reply tools/hangar/__snapshots__/test_corpus/api_path_vars.w_test_sim.md:7]: I'll leave this as a future optimization

**Pull request merged**